main.go

```diff
@@ -63,11 +65,37 @@ func RunKubeStateMetricsWrapper(opts *options.Options) {
 		factories = append(factories, crf...)
 	}
 
-	ctx := context.Background()
-	if err := app.RunKubeStateMetrics(ctx, opts, factories...); err != nil {
-		klog.ErrorS(err, "Failed to run kube-state-metrics")
-		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+	KSMRunOrDie := func(ctx context.Context) {
+		if err := app.RunKubeStateMetricsWrapper(ctx, opts, factories...); err != nil {
+			klog.ErrorS(err, "Failed to run kube-state-metrics")
+			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+		}
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	if file := options.GetOptsConfigFile(*opts); file != "" {
+		viper.SetConfigType("yaml")
+		viper.SetConfigFile(file)
+		if err := viper.ReadInConfig(); err != nil {
+			if _, ok := err.(viper.ConfigFileNotFoundError); ok {
```

Comment:
We can use `errors.Is` for better detection if it ends up being wrapped at one point.